libp2p/protocols/kademlia.nim

```diff
@@ -99,40 +100,41 @@ proc new*(
   )
 
   kad.codec = KadCodec
-  kad.handler = proc(
-      conn: Connection, proto: string
-  ) {.async: (raises: [CancelledError]).} =
-    defer:
-      await conn.close()
-    while not conn.atEof:
-      let buf =
-        try:
-          await conn.readLp(MaxMsgSize)
-        except LPStreamEOFError:
+  if not client:
+    kad.handler = proc(
```

Comment:
code all over the place does this assigns like this. would be better if there was handle function so that code is 
`kad.handler = kad.handlerImp`, this makes constructor `new` much more concise and shorter. 

then if code goes down this route, why handlers are not just some proc/method on type?